cookbooks/bcpc-hadoop/recipes/zabbix_web_monitoring.rb

```diff
@@ -0,0 +1,135 @@
+ruby_block "zabbix_web_monitoring" do
+  block do
+    require 'zabbixapi'
+
+    # Make connection to zabbix api url
+    zbx = ZabbixApi.connect(
+      :url => "https://#{node['bcpc']['management']['vip']}" +
+        ":#{node['bcpc']['zabbix']['web_port']}/api_jsonrpc.php",
+      :user => get_config!('zabbix-admin-user'),
+      :password => "#{get_config!('password','zabbix-admin','os')}"
+    )
+    if zbx.nil?
+      Chef::Log.error("Could not connect to Zabbix server")
+      raise "Could not connect to Zabbix server"
+    end
+
+    # Create zabbix host group same as the chef environment name
+    hostgroup_id = zbx.hostgroups.get_id(:name => "#{node.chef_environment}")
+    if hostgroup_id.nil?
+      hostgroup_id = zbx.hostgroups.create(:name => "#{node.chef_environment}")
+
+      # Permission Guests usergroup to read hostgroup_id
+      guests_ugroup_id = zbx.usergroups.get_id(:name => 'Guests')
+      if guests_ugroup_id.nil?
+        Chef::Log.info("Could not find 'Guests' user group in zabbix")
+      else
+        ret = zbx.usergroups.set_perms(
+                :usrgrpid => guests_ugroup_id, :hostgroupids => [hostgroup_id],
+                :permission => 2)
```

Comment:
Would you happen to have a reference to the API (e.g. what's `2` mean) which could be left as a comment for someone coming along to maintain this?